src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/PlayerBeanValidationTest.java

```diff
@@ -21,32 +28,59 @@ void setup() {
     validatorFactory.close();
   }
 
-  @Test
-  void valid_player_is_accepted() {
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("playerNames")
+  void valid_player_is_accepted(String name, ValidationListMatcher validationListMatcher) {
     Player player = new Player();
-    player.setName("test");
+    if (name != null) {
+      player.setName(name);
+    }
 
-    assertThat(validator.validate(player), empty());
+    assertThat(validator.validate(player), validationListMatcher);
   }
 
-  @Test
-  void player_with_too_short_name_is_not_valid() {
-    Player player = new Player();
-    player.setName("t");
-
-    assertThat(validator.validate(player), not(empty()));
+  private static Stream<Arguments> playerNames() {
+    return Stream.of(
+            Arguments.of("test", empty()),
+            Arguments.of(null, not(empty())),
+            Arguments.of("t", not(empty())),
+            Arguments.of("test5".repeat(10) + "a", not(empty())),
+            Arguments.of("<b>test</b>", not(empty())),
+            Arguments.of("<script>alert(1)</script>", not(empty())))
+        .map(
+            arguments ->
+                Arguments.of(arguments.get()[0], ValidationListMatcher.from(arguments.get()[1])));
   }
 
-  @Test
-  void player_with_too_long_name_is_not_valid() {
-    Player player = new Player();
-    player.setName("test5".repeat(10) + "a");
+  private interface ValidationListMatcher
+      extends Matcher<Collection<? extends ConstraintViolation<Player>>> {
 
-    assertThat(validator.validate(player), not(empty()));
-  }
+    static ValidationListMatcher from(Object object) {
+      //noinspection unchecked
+      return from((Matcher<Collection<? extends ConstraintViolation<Player>>>) object);
+    }
+
+    static ValidationListMatcher from(
+        Matcher<Collection<? extends ConstraintViolation<Player>>> source) {
+      return new ValidationListMatcher() {
+        @Override
+        public boolean matches(Object actual) {
+          return source.matches(actual);
+        }
+
+        @Override
+        public void describeMismatch(Object actual, Description mismatchDescription) {
+          source.describeMismatch(actual, mismatchDescription);
+        }
+
+        @Override
+        public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {}
 
-  @Test
-  void player_with_null_name_is_not_valid() {
-    assertThat(validator.validate(new Player()), not(empty()));
+        @Override
+        public void describeTo(Description description) {
+          source.describeTo(description);
+        }
+      };
+    }
```

Comment:
I don't really get, what additional value this test class has.

Reply:
It allows to test the bean validations without booting the whole application.